index.js

```diff
@@ -25,6 +25,31 @@ app.get('/api/v1/favorites', (request, response) => {
     });
 });
 
+app.post('/api/v1/favorites', (request, response) => {
+  const favorite = request.body;
+  for (let requiredParameter of ['name', 'artist_name', 'genre', 'rating']) {
+    if (!favorite[requiredParameter]) {
+      return response
+      .status(422)
+      .send({ error: `Expected format: { name: <String>, artist_name: <String>, genre: <String>, rating: <Integer> }. You're missing a "${requiredParameter}" property.`});
+    }
+  }
+
+  database('favorites').insert(favorite, ['id', 'name', 'artist_name', 'genre', 'rating'])
```

Comment:
This is awesome that we figured out how to return the entire object, great problem solving and teamwork!